flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -119,7 +120,7 @@ public MetricRegistryImpl(MetricRegistryConfiguration config, Collection<Reporte
 				try {
 					Optional<String> configuredPeriod = reporterSetup.getIntervalSettings();
 					TimeUnit timeunit = TimeUnit.SECONDS;
-					long period = 10;
+					long period = MetricOptions.REPORTER_INTERVAL.defaultValue().getSeconds();
```

Comment:
maybe milliseconds are more secure to use?

Reply:
unlikely; and sub-second interval will just overload whatever external system you have.

Reply:
What would be more secure about it?

Reply:
Are you referring to https://issues.apache.org/jira/browse/FLINK-18310?

Reply:
milliseconds have more precision